### PR TITLE
Feature/audio io

### PR DESCRIPTION
**Pull request by tam-nguyen-senseye**
Re-adds and refactors audio i/o modules that were previously taken out.

This will make implementing audio recording in [senseye-ptsd-runner](https://github.com/senseye-inc/senseye-ptsd-runner) much easier, since it naturally works with the Stream/StreamHandler module that's currently being used in that project.

[sc-22291]

### CODE REVIEW

**Review by Haplo-Dragon — APPROVED** [review]
Looks good! Nice that you updated the `pyproject.toml`.

**Review by bobbys-dev — APPROVED** [review]
Nice work. Thank our predecessors so that we didn't have to fully reinvent the wheel.

File: senseye_cameras/input/microphone.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    Args:
+        id (str): device index
+        config (dict): Configuration dictionary. Accepted keywords:
+            channels (int): number of audio channels
+            block_size (int): number of bytes to read in per iteration
+            samplerate (int): audio sample rate
+    '''
+
+    def __init__(self, id=0, config={}):
```
[the remark this answers is not in the document]
Comment (tam-nguyen-senseye) [reply senseye_cameras/input/microphone.py:23]: good point, will do

File: senseye_cameras/output/audio_file.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    Args:
+        path (str): Output path of video.
+        config (dict): Configuration dictionary. Accepted keywords:
+            channels (int): number of audio channels
+            samplerate (int): audio sample rate
+            subtype (str): audio type
+    '''
+
+    def __init__(self, path=None, config={}):
+        defaults = {
```
[the remark this answers is not in the document]
Comment (tam-nguyen-senseye) [reply senseye_cameras/output/audio_file.py:23]: this is from an old commit, the header now looks like this:
```
def __init__(self, **kwargs):
```

**Pull request merged**